libs/dh/charges/feature-prices/src/lib/drawer/price-tab/dh-charges-charge-prices-tab.component.ts

```diff
@@ -14,24 +14,236 @@
 })
-export class DhChargesChargePricesTabComponent {
+export class DhChargesChargePricesTabComponent
+  implements OnInit, OnChanges, OnDestroy
+{
+  @ViewChild(DhSharedUiPaginatorComponent)
+  paginator!: DhSharedUiPaginatorComponent;
+  @ViewChild(MatSort) matSort!: MatSort;
+  @ViewChild(DhDrawerDatepickerComponent)
+  drawerDatepickerComponent!: DhDrawerDatepickerComponent;
+
+  @Input() charge: ChargeV1Dto | undefined;
+  constructor(private chargePricesStore: DhChargePricesDataAccessApiStore) {}
+
+  localTimeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+
+  searchCriteria: ChargePricesSearchCriteriaV1Dto = {
+    chargeId: '',
+    fromDateTime: zonedTimeToUtc(
+      new Date(new Date().toDateString()),
+      this.localTimeZone
+    ).toISOString(),
+    toDateTime: zonedTimeToUtc(
+      new Date(new Date().toDateString()),
+      this.localTimeZone
+    ).toISOString(),
+    isDescending: false,
+    sortColumnName: SortColumnName.FromDateTime,
+    skip: 0,
+    take: 0,
+  };
+
+  result: ChargePriceV1Dto[] | undefined;
+  isLoading = this.chargePricesStore.isLoading$;
+  hasLoadingError = this.chargePricesStore.hasGeneralError$;
+  chargePricesNotFound = this.chargePricesStore.chargePricesNotFound$;
+  showDateTime = false;
+  displayedColumns = ['fromDateTime', 'time', 'price'];
+
+  private destroy$ = new Subject<void>();
+
+  readonly dataSource: MatTableDataSource<ChargePriceV1Dto> =
+    new MatTableDataSource<ChargePriceV1Dto>();
+
+  ngOnInit() {
+    this.chargePricesStore.all$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((chargePrices) => {
+        this.dataSource.data = chargePrices ?? new Array<ChargePriceV1Dto>();
+        this.result = chargePrices;
+        this.dataSource.sort = this.matSort;
+      });
+
+    this.chargePricesStore.totalAmount$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((totalAmount) => {
+        if (this.paginator) this.paginator.length = totalAmount;
+      });
+  }
+
+  ngOnChanges() {
+    if (this.charge) {
+      this.searchCriteria.chargeId = this.charge.id ?? '';
+      this.searchCriteria.skip = 0;
+
+      this.showDateTime =
+        this.charge?.resolution == Resolution.PT1H ||
+        this.charge?.resolution == Resolution.PT15M;
+
+      if (this.showDateTime)
+        this.displayedColumns = ['fromDateTime', 'time', 'price'];
+      else this.displayedColumns = ['fromDateTime', 'price'];
+
+      if (this.paginator) {
+        this.paginator.length = 0;
+        this.paginator.instance.pageIndex = 0;
+      }
+    }
+  }
+
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.unsubscribe();
+  }
+
   dateRangeChanged(dateRange: DatePickerData) {
-    console.log(dateRange);
+    this.searchCriteria.fromDateTime = zonedTimeToUtc(
+      dateRange.startDate,
+      this.localTimeZone
+    ).toISOString();
+    this.searchCriteria.toDateTime = zonedTimeToUtc(
+      dateRange.endDate,
+      this.localTimeZone
+    ).toISOString();
```

Comment:
Lines 150 - 157 are duplicated in 172 - 179. I suggest we move the code to a new method.